### PR TITLE
add types for 'draggabilly'

### PR DESCRIPTION
**Pull request by jaydubu**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/draggabilly/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+export default class Draggabilly {
+    position: Position;
+
+    constructor(element: Element | string, options?: DraggabillyOptions);
+
+    on(eventName: DraggabillyEventName, listener: (event: Event, pointer: MouseEvent | Touch, moveVector: Position) => void): Draggabilly;
+
+    off(eventName: DraggabillyEventName, listener: (event: Event, pointer: MouseEvent | Touch, moveVector: Position) => void): Draggabilly;
+
+    once(eventName: DraggabillyEventName, listener: (event: Event, pointer: MouseEvent | Touch, moveVector?: Position) => void): Draggabilly;
```
Comment (rbuckton) [inline types/draggabilly/index.d.ts]: The `?` isn't necessary on the `moveVector` parameter.

File: types/draggabilly/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+export type DraggabillyEventName = 'dragStart' | 'dragMove' | 'dragEnd' | 'pointerDown' | 'pointerMove' | 'pointerUp'
+    | 'staticClick';
+
+export default class Draggabilly {
+    position: Position;
+
+    constructor(element: Element | string, options?: DraggabillyOptions);
+
+    on(eventName: DraggabillyEventName, listener: (event: Event, pointer: MouseEvent | Touch, moveVector: Position) => void): Draggabilly;
```
Comment (rbuckton) [inline types/draggabilly/index.d.ts]: According to the documentation, the `moveVector` parameter is only supported for the `'dragMove'` and `'pointerMove'` events. Consider adding an overload for `on`, `off`, and `once`.

**Review by rbuckton — CHANGES_REQUESTED** [review]

File: types/draggabilly/index.d.ts (hunk 2 above)
Comment (jaydubu) [inline types/draggabilly/index.d.ts]: Thanks a lot for the feedback, I have implemented your suggestions by overloading the `on`, `off` and `once` functions, and removing the `?` on the `moveVector` parameter.  Please review my changes when you got a chance.  Thanks again.

**Review by rbuckton — APPROVED** [review]

**Comment by rbuckton** [comment]
Thanks for the contribution!

**Pull request merged**